### ISSUE TITLE
Can't find default framework include path on macOS

### ISSUE THREAD

**Issue by paraboul** [niosus/EasyClangComplete#626]
I get this error when calling a base class method from a children class :

```
cannot initialize object parameter of type <baseclass> with an expression of type <children class>
```

Disabling `use_libclang` fixes the issue.
No warning or error when building my project.

`.clang_complete` contains the same include path than my Makefile.

Also get various `template not found` issues (I attached a screenshot).

I can't see any error in the logs.

----------------------------------------------------

# System info: #
- Sublime Text version: `3.2.1`
- Which system are you on: `macOS 10.14.6`
- Clang version: `Apple LLVM version 10.0.1 (clang-1001.0.46.4)`

# Logs

```
[ECC:DEBUG]:[cmake_file.py]:[get_flags]:[Thread-4]: [cmake]:[get]: for file /Users/paraboul/dev/foo/foo-renderer/frontend/src/App.cpp
[ECC:DEBUG]:[cmake_file.py]:[get_flags]:[Thread-4]: [cmake]:[cached]: 'None'
[ECC:DEBUG]:[tools.py]:[search]:[Thread-4]: Searching 'CMakeLists.txt' file in: SearchScope: from_folder: /Users/paraboul/dev/foo/foo-renderer/frontend/src, to_folder: /Users/paraboul/dev/foo/foo-renderer
[ECC:DEBUG]:[cmake_file.py]:[get_flags]:[Thread-4]: No CMakeLists.txt file with 'project' in it found.
[ECC:DEBUG]:[tools.py]:[search]:[Thread-4]: Searching 'compile_commands.json' file in: SearchScope: from_folder: /Users/paraboul/dev/foo/foo-renderer/frontend/src, to_folder: /Users/paraboul/dev/foo/foo-renderer
[ECC:DEBUG]:[CppProperties.py]:[get_flags]:[Thread-4]: [CppProperties]:[get]: for file /Users/paraboul/dev/foo/foo-renderer/frontend/src/App.cpp
[ECC:DEBUG]:[CppProperties.py]:[get_flags]:[Thread-4]: [CppProperties]:[cached]: 'None'
[ECC:DEBUG]:[tools.py]:[search]:[Thread-4]: Searching 'CppProperties.json' file in: SearchScope: from_folder: /Users/paraboul/dev/foo/foo-renderer/frontend/src, to_folder: /Users/paraboul/dev/foo/foo-renderer
[ECC:DEBUG]:[c_cpp_properties.py]:[get_flags]:[Thread-4]: [c_cpp_properties]:[get]: for file /Users/paraboul/dev/foo/foo-renderer/frontend/src/App.cpp
[ECC:DEBUG]:[c_cpp_properties.py]:[get_flags]:[Thread-4]: [c_cpp_properties]:[cached]: 'None'
[ECC:DEBUG]:[tools.py]:[search]:[Thread-4]: Searching 'c_cpp_properties.json' file in: SearchScope: from_folder: /Users/paraboul/dev/foo/foo-renderer/frontend/src, to_folder: /Users/paraboul/dev/foo/foo-renderer
[ECC:DEBUG]:[flags_file.py]:[get_flags]:[Thread-4]: [clang_complete_file]:[get]: for file /Users/paraboul/dev/foo/foo-renderer/frontend/src/App.cpp
[ECC:DEBUG]:[flags_file.py]:[get_flags]:[Thread-4]: [clang_complete_file]:[cached]: '/Users/paraboul/dev/foo/foo-renderer/frontend/.clang_complete'
[ECC:DEBUG]:[tools.py]:[search]:[Thread-4]: Searching '.clang_complete' file in: SearchScope: from_folder: /Users/paraboul/dev/foo/foo-renderer/frontend/src, to_folder: /Users/paraboul/dev/foo/foo-renderer
[ECC:DEBUG]:[tools.py]:[search]:[Thread-4]: Found '.clang_complete' file: /Users/paraboul/dev/foo/foo-renderer/frontend/.clang_complete
[ECC:DEBUG]:[tools.py]:[search]:[Thread-4]: Nothing to search for in file so its ok.
[ECC:DEBUG]:[flags_file.py]:[get_flags]:[Thread-4]: [clang_complete_file]:[current]: '/Users/paraboul/dev/foo/foo-renderer/frontend/.clang_complete'
[ECC:DEBUG]:[flags_file.py]:[get_flags]:[Thread-4]: [clang_complete_file]: found cached .clang_complete
[ECC:DEBUG]:[flags_file.py]:[get_flags]:[Thread-4]: [clang_complete_file]:[unchanged]: load cached
[ECC:DEBUG]:[view_config.py]:[__load_source_flags]:[Thread-4]: flags generated from '.clang_complete'.
[ECC:DEBUG]:[view_config.py]:[__generate_essentials]:[Thread-4]: Merge
[ECC:DEBUG]:[view_config.py]:[__merge_flags]:[Thread-4]: lang flags: [(-x c++), (-std=c++11), (-I/usr/local/include), (-I/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1), (-I/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/clang/10.0.1/include), (-I/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include), (-I/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include), (-I/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.14.sdk/System/Library/Frameworks (framework directory))]
[ECC:DEBUG]:[view_config.py]:[__merge_flags]:[Thread-4]: Removing default std flag: '-std=c++11' in favor of: '-std=c++14'
[ECC:DEBUG]:[view_config.py]:[needs_update]:[Thread-4]: view config needs no update.
[ECC:DEBUG]:[view_config.py]:[update_if_needed]:[Thread-4]: config updates existing completer.
[ECC:DEBUG]:[lib_complete.py]:[update]:[Thread-4]: view is 290
[ECC:DEBUG]:[lib_complete.py]:[update]:[Thread-4]: reparsing translation_unit for view 290
[ECC:DEBUG]:[lib_complete.py]:[update]:[Thread-4]: reparsed in 0.054306983947753906 seconds
[ECC:DEBUG]:[popup_error_vis.py]:[generate]:[Thread-4]: generating error regions for view 290
[ECC:DEBUG]:[popup_error_vis.py]:[generate]:[Thread-4]: removing old error regions
[ECC:DEBUG]:[popup_error_vis.py]:[generate]:[Thread-4]: 1 error regions ready
[ECC:DEBUG]:[popup_error_vis.py]:[show_errors]:[Thread-4]: showing error regions: [(1691, 1693), (1579, 1583), (1577, 1579), (1418, 1424), (1463, 1467), (1461, 1463)]
[ECC:DEBUG]:[EasyClangComplete.py]:[config_updated]:[Thread-4]: updated config: <EasyClangComplete.plugin.view_config.ViewConfig object at 0x10417a590>
```

<img width="850" alt="Screenshot 2019-08-12 at 13 30 48" src="https://user-images.githubusercontent.com/80341/62862042-81b32c00-bd05-11e9-817e-3e62ae86c5d2.png">

<img width="1018" alt="Screenshot 2019-08-12 at 13 30 55" src="https://user-images.githubusercontent.com/80341/62862054-87107680-bd05-11e9-8416-1c2d0767e876.png">

**Comment by paraboul** [comment]
I reduced the test case.
It seems to happen whenever my base class contains a member property of type `std::unordered_map`

**Comment by paraboul** [comment]
```C++
#ifndef net_children_h__
#define net_children_h__

#include <stdint.h>
#include <unordered_map>

class Server
{
public:
    Server();
    void start();

    std::unordered_map<int, int> foo;
};

class ChildClass : public Server
{
public:
    ChildClass();
};

#endif
```

```C++
#include "test.h"

Server::Server()
{

}

void Server::start()
{

}

ChildClass::ChildClass() : Server()
{

}
```

```
    ChildClass *cc = new ChildClass();
    cc->start(); // Error (cannot initialize object ....)
 ```

**Comment by paraboul** [comment]
Reducing again.
This apparently happen only if I include this file `https://github.com/glfw/glfw/blob/master/include/GLFW/glfw3.h` before other includes.

I'm not sure to understand how this could create this error though (which only trigger with ECC libclang output, not during actual compilation).

**Comment by niosus** [comment]
This might be due to the standard library picked by lib clang. Could you maybe create a small example that I could run? This would help me try to debug this. Also, I am away from the internet for around 1.5 weeks, so I guess I will only be able to have a look later.

**Comment by paraboul** [comment]
Ok, I think I found the issue.

It seems that this `glfw3.h` includes `OpenGL/gl3.h`. This lib can't be found by libclang and creates an hidden error inside this file (failing to include subsequent headers files).

The reason libclang can't find this opengl header is because the way ECC injects default includes :

`clang++ -c /tmp/test.cc -v`

If you look at the output :

```
#include <...> search starts here:
 /usr/local/include
 /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1
 /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/clang/10.0.1/include
 /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include
 /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.14.sdk/usr/include
 /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.14.sdk/System/Library/Frameworks (framework directory)
```

For some reason Apple clang's decided to append `(framework directory)` at the end of the framework path 😒. (OpenGL headers is located in the `[...]Library/Frameworks` directory).

This is getting added by ECC :

```
[ECC:DEBUG]:[view_config.py]:[__merge_flags]:[Thread-4]: lang flags: [(-x c++), (-std=c++11), (-I/usr/local/include), (-I/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1), (-I/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/clang/10.0.1/include), (-I/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include), (-I/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include), (-I/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.14.sdk/System/Library/Frameworks (framework directory))]
```

Now I guess that we need a proper way to sanitize this...

**Comment by paraboul** [comment]
And it's actually worse than that.

`OpenGL/gl.h` should resolves to `/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/System/Library/Frameworks/OpenGL.framework/Headers/`

Looks like eclipse faced the same issue :

https://bugs.eclipse.org/bugs/show_bug.cgi?id=69529

And netbeans too :

https://netbeans.org/bugzilla/show_bug.cgi?id=160043

**Comment by paraboul** [comment]
So apparently clang and gcc provides a `-F` option to substitute `-I` in order to deal with frameworks directory.

My guess is that we should look for clang this `(framework directory)` string and use it as a hint to use the specified directory with `-F` rather than `-I`

**Comment by niosus** [comment]
Sorry, I still can't properly find the time to look into it. But it's still on my desk. If you can look into it and create a PR, that would make my life easier :)

**Comment by saagarjha** [comment]
https://github.com/niosus/EasyClangComplete/commit/18ce875fd9fecd0f97850a3d3d62647ef0c90013 doesn't fix this, unfortunately: see #652.
